Codecademy/gamut
# fix(Checkbox): increase contrast and fix VO for checkbox labels in fi…

…refox

### Overview


a11y fixes for Checkbox: increase color contrast ratio and fix voiceover for labels that include an anchor in firefox


NOTES:

**changes to colors:**
- changed checkbox outline and text color to navy-600
- changed checkbox outline and text color to navy-500 when the checkbox is disabled, and changed the text from italic to regular

before:
<img width="1230" alt="Screen Shot 2022-12-05 at 3 09 49 PM" src="https://user-images.githubusercontent.com/3805607/205733407-d52afe62-a3d3-4d19-a92e-12605d8b7c19.png">

after:
<img width="1238" alt="Screen Shot 2022-12-05 at 3 09 56 PM" src="https://user-images.githubusercontent.com/3805607/205733432-347749db-6107-4364-aacf-637b22d8cc08.png">


**changes for voiceover:**
- before, in firefox, if you had a checkbox label that included an anchor, the voiceover in firefox would read the anchor as a "group" instead of a "link" (this worked fine in chrome)
- now, in firefox, if you add the `dontAriaHideLabel` prop to your checkbox and make sure the non anchor elements within the label have aria-hidden, then voiceover will correctly read the anchor as a "link". this is because it removes the `aria-hidden` from the `CheckboxText` element, which was causing firefox to consider a link inside to be a "group" instead of a "link".
- you do have to make sure the non anchor elements have `aria-hidden`, because otherwise, the voiceover will read the label twice, first when you are on the checkbox, and then when you control+option+arrow to the label itself (which is the a11y bug that we had fixed [previously](https://github.com/Codecademy/gamut/commit/4ba01e73f10ab55bd769e2b9246bf3cd497f77c7) by putting the `aria-hidden` on the `CheckboxText` element)

before:
<img width="511" alt="Screen Shot 2022-12-05 at 3 16 47 PM" src="https://user-images.githubusercontent.com/3805607/205734522-2a8a4c46-fb0d-4124-a951-c2406d1b406b.png">


![ezgif-1-7b17017371](https://user-images.githubusercontent.com/3805607/205733634-414b9d71-95b1-4dcc-a7d7-4d238eae42b4.gif)


after:
<img width="487" alt="Screen Shot 2022-12-05 at 2 55 53 PM" src="https://user-images.githubusercontent.com/3805607/205733714-cae1b3c4-d4ba-4c6b-b875-bb63d8272f2e.png">

![ezgif-1-baca6e5931](https://user-images.githubusercontent.com/3805607/205733669-4886d119-18c5-4084-b676-0bc073209d24.gif)


### PR Checklist

- [ ] Related to JIRA ticket: [EGG-2981](https://codecademy.atlassian.net/browse/EGG-2981) and [EGG-2978](https://codecademy.atlassian.net/browse/EGG-2978)

#### Testing Instructions
1. open storybook and turn on voiceover
2. make sure last checkbox in last story ("Labels as ReactNodes") the anchor is narrated as a link in firefox, not as a group

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/33801)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
